### PR TITLE
Implement simple DNS name lookup cache

### PR DESCRIPTION
**Pull request by fholzhauser**

### CODE REVIEW

File: src/ergw_sup.erl [hunk 1]
```diff
@@ -30,6 +30,7 @@ start_link() ->
 %% ===================================================================
 
 init([]) ->
+    ok = ergw_node_selection_cache:init(),
```
Comment (vkatsuba) [inline src/ergw_sup.erl:33]: Not sure but maybe make sense create `ergw_node_selection_cache` as worker

Comment (fholzhauser) [reply src/ergw_sup.erl:33]: That was the initial implementation indeed. However I wanted to benefit from the ets concurrency for lookup requests without serialization, it ended up being an empty gen_server process just holding the ETS table without any added value.

**Review by ebengt — APPROVED** [review]
Good.
For the test you might want to consider adding meck to inet_res:resolve/4, and check that there are only 2 calls.

**Comment by vkatsuba** [comment]
Please, do squash commits in single commit.

**Review by vkatsuba — APPROVED** [review]

**Comment by fholzhauser** [comment]
Changed testcase to do more requests and count the resolve meck calls (thanks Bengt for the idea)

**Pull request merged**